api.ts

```diff
@@ -1108,6 +1110,49 @@ export interface SyncUpdateResultMarshaled<
  */
 export type GenericSyncUpdateResultMarshaled = SyncUpdateResultMarshaled<any, any, any>;
 
+/**
+ * Sets a limit on the number of permissions that can be returned for a single row
+ * in a call to {@link executeGetPermissions}.
+ *
+ * @hidden
+ */
+const MaxPermissionsPerRow = 1000;
+
+/**
+ * Type definition for the result of calls to {@link executeGetPermissions}.
+ * @hidden
+ */
+export interface ExecuteUpdatePermissionResult {
+  /**
+   * The list of permissions applying to the passed in parameters.
+   */
+  permissions: Permission[];
+}
+
+/**
+ * Type definition for the data passed to the {@link executeGetPermissions} function of a sync table.
+ *
+ * TODO(sam): Unhide this
+ * @hidden
+ */
+export interface ExecuteGetPermissionsRequest<
+  K extends string,
+  L extends string,
+  SchemaT extends ObjectSchemaDefinition<K, L>,
+> {
+  /**
+   * The list of rows for which to fetch permissions. These rows have been retrieved from the {@link execute} function
+   */
+  rows: Array<ObjectSchemaDefinitionType<K, L, SchemaT>>;
```

Comment:
I think for executeUpdate the input is actually an array of wrapper objects, rather than a wrapper object containing an array, if we're trying to model the exact same thing.

Reply:
Sure and each wrapper object contains a row in this scenario + whatever else we need to add here? 

Should we have another level of wrapping in case we want to pass in global stuff to this function or is the assumption that this will always come from the context?

Reply:
Was thinking maybe we have
```
export interface ExecuteGetPermissionRequestRow<
  K extends string,
  L extends string,
  SchemaT extends ObjectSchemaDefinition<K, L>,
>{
 row: ObjectSchemaDefinitionType<K, L, SchemaT>
 otherRowSpecificStuff: {};
}


/**
 * Type definition for the data passed to the {@link executeGetPermissions} function of a sync table.
 *
 * TODO(sam): Unhide this
 * @hidden
 */
export interface ExecuteGetPermissionsRequest<
  K extends string,
  L extends string,
  SchemaT extends ObjectSchemaDefinition<K, L>,
> {
  /**
   * The list of rows for which to fetch permissions. These rows have been retrieved from the {@link execute} function
   */
  rows: ExecuteGetPermissionRequestRow<K,L,SchemaT>[];

 otherGlobalStuff: {};
}


```

Reply:
(Mentioned on Slack as well) Yeah this double-wrapping is the most future-proof. The outer wrapping object is not consistent with executeUpdate, not sure if we care and why that wasn't done there. It's certainly verbose while we don't have any other fields, but it's certainly the most future-proof, so I'm supportive.

Reply:
Cool just pushed this